modules/core/04-channel/types/timeout_test.go

```diff
@@ -0,0 +1,109 @@
+package types_test
+
+import (
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	"github.com/cosmos/ibc-go/v7/modules/core/04-channel/types"
+)
+
+func (suite *TypesTestSuite) TestTimeoutPassed() {
+	var timeout types.Timeout
+	var passed bool
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"client height is not after timeout height",
+			func() {
+				passed = timeout.AfterHeight(clienttypes.NewHeight(0, 75))
+			},
+			false,
+		},
+		{
+			"client timestamp is not after timeout timestamp",
+			func() {
+				passed = timeout.AfterTimestamp(75)
+			},
+			false,
+		},
+		{
+			"client height is after timeout height",
+			func() {
+				passed = timeout.AfterHeight(clienttypes.NewHeight(0, 25))
+			},
+			true,
+		},
+		{
+			"client timestamp is after timeout timestamp",
+			func() {
+				passed = timeout.AfterTimestamp(25)
+			},
+			true,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			timeout = types.NewTimeout(
+				clienttypes.NewHeight(0, 50),
+				50,
+			)
+
+			tc.malleate()
+
+			if tc.expPass {
+				suite.Require().True(passed)
+			} else {
+				suite.Require().False(passed)
+			}
+		})
+	}
+}
+
+func (suite *TypesTestSuite) TestTimeout() {
```

Comment:
Maybe something like:

```suggestion
func (suite *TypesTestSuite) TestTimeoutIsValid() {
```

side note: but this looks like its being covered as part of testing `Upgrade.ValidateBasic()` but I'm happy to keep this also.

Reply:
yea, i thought about this but also thought it would make more sense to have the unit testing more explicitly associated with what is being tested, rather than as part of the tests for another method.